app/models/upload.rb

```diff
@@ -91,43 +89,9 @@ def self.reset_in_use_callbacks
   end
 
   def self.with_no_non_post_relations
-    scope = self
-      .joins(<<~SQL)
-        LEFT JOIN site_settings ss
-        ON NULLIF(ss.value, '')::integer = uploads.id
-        AND ss.data_type = #{SiteSettings::TypeSupervisor.types[:upload].to_i}
-      SQL
-      .where("ss.value IS NULL")
-      .joins("LEFT JOIN users u ON u.uploaded_avatar_id = uploads.id")
-      .where("u.uploaded_avatar_id IS NULL")
-      .joins("LEFT JOIN user_avatars ua ON ua.gravatar_upload_id = uploads.id OR ua.custom_upload_id = uploads.id")
-      .where("ua.gravatar_upload_id IS NULL AND ua.custom_upload_id IS NULL")
-      .joins("LEFT JOIN user_profiles up ON up.profile_background_upload_id = uploads.id OR up.card_background_upload_id = uploads.id")
-      .where("up.profile_background_upload_id IS NULL AND up.card_background_upload_id IS NULL")
-      .joins("LEFT JOIN categories c ON c.uploaded_logo_id = uploads.id OR c.uploaded_background_id = uploads.id")
-      .where("c.uploaded_logo_id IS NULL AND c.uploaded_background_id IS NULL")
-      .joins("LEFT JOIN custom_emojis ce ON ce.upload_id = uploads.id")
-      .where("ce.upload_id IS NULL")
-      .joins("LEFT JOIN theme_fields tf ON tf.upload_id = uploads.id")
-      .where("tf.upload_id IS NULL")
-      .joins("LEFT JOIN user_exports ue ON ue.upload_id = uploads.id")
-      .where("ue.upload_id IS NULL")
-      .joins("LEFT JOIN groups g ON g.flair_upload_id = uploads.id")
-      .where("g.flair_upload_id IS NULL")
-      .joins("LEFT JOIN badges b ON b.image_upload_id = uploads.id")
-      .where("b.image_upload_id IS NULL")
-      .joins(<<~SQL)
-        LEFT JOIN theme_settings ts
-        ON NULLIF(ts.value, '')::integer = uploads.id
-        AND ts.data_type = #{ThemeSetting.types[:upload].to_i}
-      SQL
-      .where("ts.value IS NULL")
-
-    if SiteSetting.selectable_avatars.present?
-      scope = scope.where.not(id: SiteSetting.selectable_avatars.map(&:id))
-    end
-
-    scope
+    self
```

Comment:
Have we got specs that cover this change? It seems too simple compared to the original query but maybe it's just appearances and it's fine, just want to double check.

Reply:
Yes, this part is very well covered. I did not add any new tests for this part because there is no change in functionality.

Reply:
Okay cool just thought I would double check.